### DISCUSSION TITLE
Allow access to an API endpoint for creating and managing repositories

### DISCUSSION THREAD

**Discussion by codycooperross** [datacite/datacite-suggestions#17] [General Suggestions]
### What is the problem that your suggestion solves?

Users have to manually create and update repositories using Fabrica. 

### What solution might meet your needs?

Creating and managing repositories via the REST API. 

### What alternatives have you tried or considered?

Using Fabrica to manually create repositories and see their automatically assigned prefixes. 

### Is there anything else you would like to share?

https://datacite.slack.com/archives/C07CQJ89WLT/p1721684743758199

### What group(s) would benefit from your suggestion?

- [ ] DataCite members registering DOIs and metadata and administering accounts
- [X] Developers building integrations with DataCite services
- [ ] Harvesters who index DataCite DOIs and metadata
- [ ] Individuals or institutions analyzing DataCite DOIs and metadata
- [ ] DataCite staff
- [ ] Other

**Comment by rebeccaringuette** [comment]
YES PLEASE! Especially including mass updates, e.g. via an updated mapping to DataCite implemented in custom code.

**Reply by codycooperross** [reply]
Thanks for this, Rebecca! Could you describe why you might need to perform mass updates on repository metadata?

**Reply by rebeccaringuette** [reply]
Sure. Adding a publisher ROR once it is created, adding affiliation RORs for a large number of records, and similar items. I am part of a NASA SMD data repository and we are preparing to do several mass updates. Do I understand the issue correctly?

**Reply by KellyStathis** [reply]
Hi @rebeccaringuette, you can perform these updates using our REST API by making requests to the /dois endpoint. Guidance is available in our support docs: [Updating DOIs with the REST API](https://support.datacite.org/docs/updating-metadata-with-the-rest-api) and via our API Reference: https://support.datacite.org/reference/put_dois-id. 

This suggestion is about [creating new Repository accounts](https://support.datacite.org/docs/fabrica-create-repository) and [updating Repository settings](https://support.datacite.org/docs/fabrica-update-repository), functionality which is currently only available through the DataCite Fabrica web interface.

**Comment by rebeccaringuette** [comment]
oh got it. Thanks.
Mass updates on repository metadata might be needed if a group of
repositories are managed by a team, and some information for the group
needs to be updated.

**Comment by mpfl** [comment]
I am Matthias Liffers, Product Manager of RAiD at the Australian Research Data Commons.

(context for those not familiar with RAID) RAiD (Research Activity Identifier) is built on DataCite DOIs. For each RAiD, there is a DataCite DOI that contains a subset of RAiD metadata. Each RAiD "service point" has a corresponding DataCite DOI repository.

When a new RAiD Service Point is created, a corresponding DataCite DOI Repository must be created. The creation of a single repository takes time to create the repository, set a password, and retrieve the DOI prefix as each of these operations take place on separate pages in Fabrica. The information about the repository (ID, password, prefix) is then manually entered into RAiD.

It would be great if we could automate this process. We anticipate dozens if not hundreds of new RAiD service points in the coming twelve months.

**Reply by codycooperross** [reply]
Thanks, @mpfl ! We know this is important for RAiD as RAiD service points scale. We’ll continue to follow this thread and share updates in the event of relevant developments.

**Reply by sheilarabun** [reply]
I second Matthias on this one - I am part of the team running the US RAiD Pilot and we have at least 13 repositories that we need to create for this purpose so far, with more to come. Would be great to have the ability to automate the creation of new repositories under our RAiD RA. Thank you!
